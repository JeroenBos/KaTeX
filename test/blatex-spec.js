/* eslint-disable max-len */
import {assertNodeType} from "../src/parseNode";
import parseTree from "../src/parseTree";
import {nonstrictSettings} from "./helpers";
import katex from "../katex";

describe("\\blatex{}", function() {
    it("should parse without arg", function() {
        const tree = parseTree("\\blatex{}", nonstrictSettings);

        expect(tree.length).toBe(1);
        const root = assertNodeType(tree[0], "blatex");
        expect(root.type).toBe("blatex");
        expect(root.loc.start).toBe(0);
        expect(root.loc.end).toBe(7);
        expect(root.args.length).toBe(1);
        expect(root.args[0].type).toBe("raw");
        expect(root.args[0].string).toBe("");
    });
    it("should parse with arg", function() {
        const tree = parseTree("\\blatex{arg}", nonstrictSettings);

        expect(tree.length).toBe(1);
        const root = assertNodeType(tree[0], "blatex");
        expect(root.type).toBe("blatex");
        expect(root.loc.start).toBe(0);
        expect(root.loc.end).toBe(7); // hm ideally more like 12
        expect(root.args.length).toBe(1);
        expect(root.args[0].type).toBe("raw");
        expect(root.args[0].string).toBe("arg");
    });
    it("should parse with arg and optional arg", function() {
        const tree = parseTree("\\blatex[optArg]{arg}", nonstrictSettings);

        expect(tree.length).toBe(1);
        const root = assertNodeType(tree[0], "blatex");
        expect(root.type).toBe("blatex");
        expect(root.loc.start).toBe(0);
        expect(root.loc.end).toBe(7);
        expect(root.args.length).toBe(2);
        expect(root.args[1].type).toBe("raw");
        expect(root.args[1].string).toBe("optArg");
    });

    it("renders arg", function() {
        const html = katex.renderToString("\\blatex{arg}");
        expect(html).toMatchSnapshot();
    });
    it("renders arg and optional arg", function() {
        const html = katex.renderToString("\\blatex[optArg]{arg}");
        expect(html).toMatchSnapshot();
    });
    it("renders mathematical arg", function() {
        const html = katex.renderToString("\\blatex{\\sqrt{x}}");
        expect(html).toMatchSnapshot();
    });
});
