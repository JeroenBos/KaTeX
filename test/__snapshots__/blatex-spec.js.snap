// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\\blatex{} renders arg 1`] = `
<span class="katex">
  <span class="katex-mathml">
    <math xmlns="http://www.w3.org/1998/Math/MathML">
      <semantics>
        <mrow>
          <annotation>
            arg
          </annotation>
        </mrow>
        <annotation encoding="application/x-tex">
          \\blatex{arg}
        </annotation>
      </semantics>
    </math>
  </span>
  <span class="katex-html"
        aria-hidden="true"
  >
    <span class="base">
      <span class="strut"
            style="height:0.4306em;"
      >
      </span>
      <span data-loc="0,12">
        arg
      </span>
    </span>
  </span>
</span>
`;

exports[`\\blatex{} renders arg and optional arg 1`] = `
<span class="katex">
  <span class="katex-mathml">
    <math xmlns="http://www.w3.org/1998/Math/MathML">
      <semantics>
        <mrow>
          <annotation>
            argoptArg
          </annotation>
        </mrow>
        <annotation encoding="application/x-tex">
          \\blatex[optArg]{arg}
        </annotation>
      </semantics>
    </math>
  </span>
  <span class="katex-html"
        aria-hidden="true"
  >
    <span class="base">
      <span class="strut"
            style="height:0.4306em;"
      >
      </span>
      <span data-blatex="optArg"
            data-loc="0,20"
      >
        arg
      </span>
    </span>
  </span>
</span>
`;

exports[`\\blatex{} renders mathematical arg 1`] = `
<span class="katex">
  <span class="katex-mathml">
    <math xmlns="http://www.w3.org/1998/Math/MathML">
      <semantics>
        <mrow>
          <annotation>
            \\sqrt{x}
          </annotation>
        </mrow>
        <annotation encoding="application/x-tex">
          \\blatex{\\sqrt{x}}
        </annotation>
      </semantics>
    </math>
  </span>
  <span class="katex-html"
        aria-hidden="true"
  >
    <span class="base">
      <span class="strut"
            style="height:1em;vertical-align:-0.25em;"
      >
      </span>
      <span data-loc="0,17">
        \\sqrt{x}
      </span>
    </span>
  </span>
</span>
`;
